core/src/utils/optional.hpp

```diff
@@ -101,7 +101,7 @@
 
 namespace std{
 
-    namespace experimental{
+    namespace ledger_exp{
```

Comment:
I think we should remove that from `std` if we decide to change the namespace. I’m not comfortable with intruding in other namespaces… 😕

Reply:
Yeah, you right. 
I will try to completely move to std::optional.
Looks like it should be safe, I will upgrade compilers on CI in coming PR
https://en.cppreference.com/w/cpp/compiler_support

Reply:
@phaazon I will do this change in two steps. First I would like temporary merge this commit, it will allow me safely upgrade compilers on CI. Second step I will switch to std::optional.
This way will be easy to understand what exactly cause problems, if they happen.